Automattic/simplenote-macos
# Add lane to deploy to App Store and publish to GitHub

Originally, I had a CI part in this PR, too. I removed it to implement the API-driven deployment as opposed to the tag-driven workflow (see https://github.com/Automattic/simplenote-ios/pull/1214) instead.

Note: I'm pushing this to `release/2.10` in the hope that I'll be able to benefit of these improvements when it comes the time to make the build for the App Store.

### Test

- Checkout this PR and create a new branch from it
- Bump the `BUILD_NUMBER` in `config/Version.public.xcconfig` otherwise ASC will reject your build (via email, the upload will still work, unlike what happens with TestFlight 🙄 ) – ~~I run my test with `11227`, so the first reviewer should use `11228`, the second `11229` and so on...~~ The build number from #932 is `11230` so the first person reviewing should use `11231`, if there's a second reviewer, they should use `11232`, and so on...
- Run `bundle exec fastlane run configure_apply` to get the latest version of the secrets
- Add two new keys to your to your environment file `~/.simplenotemacos-env.default`,  `APP_STORE_CONNECT_API_KEY_PATH` and `SIGH_API_KEY_PATH` both with value `".configure-files/app_store_connect_fastlane_api_key.json"` see `user.env-example` for more info.
- `bundle install && bundle exec fastlane build_and_upload_app_store create_github_release:true`

It should succeed, upload your build to App Store Connect, and create a new release in this repo.

<table>
<tr>
<td>

![image](https://user-images.githubusercontent.com/1218433/112789473-e4a73180-90a8-11eb-900c-0c0d012ea0a6.png)

</td>
<td>

![image](https://user-images.githubusercontent.com/1218433/112588837-30ad6880-8e54-11eb-9625-a749dcc6c938.png)

</td>
</tr>
</table>

Once verified:

- Delete your test branch
- Delete the release from GitHub

### Review
Only one developer required to review these changes, but anyone can perform the review.

### Release
These changes do not require release notes.
